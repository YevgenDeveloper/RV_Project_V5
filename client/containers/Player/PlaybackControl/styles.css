@import 'client/styles/variables';

.control {
  display: flex;
  flex-flow: row nowrap;
  margin-right: 10px;
}

.button {
  width: 44px;
  overflow: hidden;
  text-indent: -1000px;
  margin-right: 1px;
  position: relative;

  &:first-child {
    border-radius: 4px 0 0 4px;
  }

  &.on {
    background-color: $color-blue;

    &:hover:enabled {
      background-color: color-mod($color-blue l(48%));
    }

    &:active:enabled {
      background-color: color-mod($color-blue l(44%));
    }
  }

  &:last-child {
    border-radius: 0 4px 4px 0;
    margin-left: 0;
  }

  &.playBack::after {
    background-image: url('client/assets/icn-play-back.svg');
    background-position: 45% 50%;
  }

  &.playForward::after {
    background-image: url('client/assets/icn-play-forward.svg');
    background-position: 55% 50%;
  }

  &.on.playBack::after {
    background-image: url('client/assets/icn-play-back-white.svg');
  }
  &.on.playForward::after {
    background-image: url('client/assets/icn-play-forward-white.svg');
  }
}
