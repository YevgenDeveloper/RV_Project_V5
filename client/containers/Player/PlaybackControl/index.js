// @flow

import React from 'react';
import classNames from 'classnames/bind';

import { playModes } from 'client/constants';
import { Button } from 'client/components';
import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  disabledBack: boolean,
  disabledForward: boolean,
  disabled: boolean,
  playMode: string,
  onClickPlayBack: void => void,
  onClickPlayForward: void => void,
};

const PlaybackControl = (props: Props) => (
  <div className={styles.control}>
    <Button
      label="Play Back"
      onClick={props.onClickPlayBack}
      className={cx(styles.button, styles.playBack, {
        [styles.on]: props.playMode === playModes.BACKWARD,
      })}
      disabled={props.disabledBack || props.disabled}
    />
    <Button
      label="Play Forward"
      onClick={props.onClickPlayForward}
      className={cx(styles.button, styles.playForward, {
        [styles.on]: props.playMode === playModes.FORWARD,
      })}
      disabled={props.disabledForward || props.disabled}
    />
  </div>
);

export default PlaybackControl;
