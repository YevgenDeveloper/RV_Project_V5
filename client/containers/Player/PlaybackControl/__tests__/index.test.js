// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import PlaybackControl from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(
    <PlaybackControl
      disabledBack={false}
      disabledForward={false}
      disabled={false}
      onClickPlayBack={jest.fn()}
      onClickPlayForward={jest.fn()}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
