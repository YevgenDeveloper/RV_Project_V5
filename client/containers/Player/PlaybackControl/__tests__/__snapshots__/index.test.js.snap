// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlaybackControl
    disabled={false}
    disabledBack={false}
    disabledForward={false}
    onClickPlayBack={[MockFunction]}
    onClickPlayForward={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          blue={false}
          className="button playBack"
          dark={false}
          disabled={false}
          label="Play Back"
          onClick={[MockFunction]}
        />,
        <Button
          blue={false}
          className="button playForward"
          dark={false}
          disabled={false}
          label="Play Forward"
          onClick={[MockFunction]}
        />,
      ],
      "className": "control",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button playBack",
          "dark": false,
          "disabled": false,
          "label": "Play Back",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button playForward",
          "dark": false,
          "disabled": false,
          "label": "Play Forward",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            blue={false}
            className="button playBack"
            dark={false}
            disabled={false}
            label="Play Back"
            onClick={[MockFunction]}
          />,
          <Button
            blue={false}
            className="button playForward"
            dark={false}
            disabled={false}
            label="Play Forward"
            onClick={[MockFunction]}
          />,
        ],
        "className": "control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button playBack",
            "dark": false,
            "disabled": false,
            "label": "Play Back",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button playForward",
            "dark": false,
            "disabled": false,
            "label": "Play Forward",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
